### PR TITLE
feat(FIR-25191): Add option to sort result lines before comparison

### PR DESCRIPTION
**Pull request by leonard-firebolt**
## Summary:

**TL;DR:** This PR makes it possible to write SQL tests for queries without an ORDER BY clause, even if their output row order is not deterministic. That's achieved by lexicographically sorting the output rows.

If a query does not use an ORDER BY clause, the order of the result rows is unspecified. Adding an ORDER BY clause is an option to get deterministic output. However, now you are testing a different query.
This PR adds the option `sort_rows` to our SQL test framework. If set, the test driver sorts the expected and actual output rows lexicographically before comparison.

## Test Plan:

Added an example SQL test to `sample.test`:

```
# If you specify sort_rows, the actual and expected result rows are sorted lexicographically before comparison.
[sort_rows]
[config=force_pgdate_timestampntz=1,current_timestamptz={{2023-02-06T12:34:56|2023-05-06T12:34:56}}]
SELECT current_date AS d, NULL AS ntz, NULL AS tz
UNION ALL
SELECT NULL AS d, localtimestamp()::TIMESTAMPTZ AS ntz, NULL AS tz
UNION ALL
SELECT NULL AS d, NULL AS ntz, current_timestamp::TIMESTAMPTZ AS ts;
--
ALTERNATION GROUP: 2023-02-06T12:34:56
--
d PGDATE,ntz TIMESTAMPTZ,tz TIMESTAMPTZ
NULL,'2023-02-06 12:34:56+00',NULL
NULL,NULL,'2023-02-06 12:34:56+00'
'2023-02-06',NULL,NULL
--
ALTERNATION GROUP: 2023-05-06T12:34:56
--
d PGDATE,ntz TIMESTAMPTZ,tz TIMESTAMPTZ
NULL,NULL,'2023-05-06 12:34:56+00'
'2023-05-06',NULL,NULL
NULL,'2023-05-06 12:34:56+00',NULL
==
```

Also added a unit test to `file_based_test_driver_test.cc`:

```
TEST_TMPDIR="/tmp" /home/leonard/code/packdb/build/Debug/contrib/file-based-test-driver-cmake/file_based_test_driver_tests --gtest_filter="*SortResultLinesBeforeComparison*"
Running main() from /home/leonard/code/packdb/contrib/googletest/googletest/src/gtest_main.cc
Note: Google Test filter = *SortResultLinesBeforeComparison*
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from TestdataUtilCallbackTest
[ RUN      ] TestdataUtilCallbackTest.SortResultLinesBeforeComparison
[       OK ] TestdataUtilCallbackTest.SortResultLinesBeforeComparison (1 ms)
[----------] 1 test from TestdataUtilCallbackTest (1 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (1 ms total)
[  PASSED  ] 1 test.
```

### CODE REVIEW

File: file_based_test_driver/file_based_test_driver.cc [hunk 1]
```diff
@@ -303,7 +303,55 @@ static bool CompareAndAppendOutput(
     std::vector<TestCasePartComments>* comments, std::string* all_output,
     bool expected_output_is_regex = false,
     const std::vector<std::string> expected_parts = {},
-    const std::vector<std::string> output_parts = {}) {
+    const std::vector<std::string> output_parts = {},
+    bool sort_result_lines = false) {
+  // Firebolt Start
+  // If sort_result_lines is true, sort the lines of the actual and expected
+  // result and call CompareAndAppendOutput again with the modified arguments.
+  if (sort_result_lines && expected_parts.size() == output_parts.size()) {
```
Comment (moshap-firebolt) [inline file_based_test_driver/file_based_test_driver.cc:311]: can you add a test case to file_based_driver to demonstrate how it works

Comment (leonard-firebolt) [reply file_based_test_driver/file_based_test_driver.cc:311]: Pasted the output of the test into the PR description

File: file_based_test_driver/file_based_test_driver_test.cc [hunk 2]
```diff
@@ -36,9 +36,9 @@
 #include "file_based_test_driver/base/status_matchers.h"
 
 ABSL_DECLARE_FLAG(int32_t, file_based_test_driver_insert_leading_blank_lines);
-ABSL_DECLARE_FLAG(bool, file_based_test_driver_generate_test_output);
```
[the remark this answers is not in the document]
Comment (leonard-firebolt) [reply file_based_test_driver/file_based_test_driver_test.cc:39]: It's removed because it's declared in `file_based_test_driver_test.cc` but it's not defined anywhere. Compiler complained about missing definition.

File: file_based_test_driver/run_test_case_result.h [hunk 3]
```diff
@@ -76,6 +76,11 @@ class RunTestCaseResultBase {
   void set_expected_output_is_regex(bool expected_output_is_regex) {
     expected_output_is_regex_ = expected_output_is_regex;
   }
+
+  bool sort_result_lines() const { return sort_result_lines_; }
```
[the remark this answers is not in the document]
Comment (leonard-firebolt) [reply file_based_test_driver/run_test_case_result.h]: Done
